src/pevm.rs

```diff
@@ -458,6 +461,19 @@ fn transform_output(
                 cumulative_gas_used,
                 logs: result.into_logs(),
             };
+            let state: EvmStateTransitions = state
+                .into_iter()
+                .filter(|(_, account)| account.is_touched())
+                .map(|(address, account)| {
+                    if account.is_selfdestructed()
+                        || account.is_empty() && spec_id.is_enabled_in(SpecId::SPURIOUS_DRAGON)
```

Comment:
Let's link to EIP-161 since this condition is not as self-described as `is_selfdestructed`.